app/src/main/java/protect/card_locker/ImportExportActivity.java

```diff
@@ -419,4 +424,31 @@ private void onExportComplete(ImportExportResult result, final Uri path) {
 
         builder.create().show();
     }
+
+    private void handleImportingFileOpenedByIntentFilter(Intent fileIntent) {
+        try {
+            String intentType = fileIntent.getType();
+            if (intentType.contains("/zip")) {
+                Uri fileData = fileIntent.getData();
+                String filePath = fileData.getPath();
+                if (filePath.contains("fidme-export-request")) {
+                    importDataFormat = DataFormat.Fidme;
+                } else if (filePath.contains("sync")) {
+                    importDataFormat = DataFormat.Stocard;
+                } else {
+                    importDataFormat = DataFormat.Catima;
+                }
+            } else if (intentType.contains("/comma-separated-values")) {
+                importDataFormat = DataFormat.Catima;
+            } else if (intentType.contains("/json")) {
+                importDataFormat = DataFormat.VoucherVault;
+            } else {
+                throw new IllegalArgumentException("Unknown DataFormat");
+            }
+            openFileForImport(fileIntent.getData(), null);
+        } catch (IllegalArgumentException e) {
+            Toast.makeText(getApplicationContext(), R.string.failedOpeningFileManager, Toast.LENGTH_LONG).show();
```

Comment:
@TheLastProject - I was looking for a constants file in the project and did not find one and wanted to know if you would prefer that I move the string literals here into constants?

Reply:
Hmm, I'm actually kinda worried with how this works. Opening a file will now instantly import with no user interaction, making it very easy to accidentally make a mess of everything. Maybe it's better to just open the list of supported importers and let the user choose (they'll likely know what kind of backup they're trying to import anyway). I think my idea of "Catima should select the correct importer based on the filename" was actually a bad idea seeing this.

I'm also a bit worried about the flow here as it's setting a variable that `openFileForImport` uses implicitly instead of something passed to the function explicitly. I see that that was part of the original design already though, so fixing that is probably out of scope (although probably should be done at some point, just like how the ImportExportActivity should be changed to first select a file and then the importer to prevent confusion for the use file manager and use other app being the same).

Reply:
@TheLastProject - Just so we don't run into another confusion, you would prefer the intent filters to trigger the same logic as the **Import From Filesystem button**? Meaning, the dialog opens up and let's the user select which file he/she is going to import.

A small heads up as there will be a weird flow there.
When the  Import From Filesystem button is clicked inside the application, after a user selects the type of file to import, he/she is then redirected to the file manager to choose the file.
When a file is opened from outside the application, you do not want the same user flow to happen since the user has just selected the file.

I hope that makes sense.

Reply:
I think what we want for the "starting outside" flow is:

1. User opens File Manager
2. User clicks file there
3. Catima opens and asks what type the import is (Catima/Stocard/Fidme/etc.)
4. User selects option and import starts with file selected in step 2

---

_Sidenote:_
I know that is slightly different from what Catima's in-app flow currently is:
1. User opens Import/Export screen
2. User clicks Import From Filesystem
3. User is asked what type the import is
4. User selects file
5. User selects option and import starts

At some point I want to change the in-app flow to:
1. User opens Import/Export screen
2. User clicks Import From Filesystem
3. User selects file
4. User is asked what type the import is
5. User selects option and import starts

But that feels out of scope for your change (although if you want to implement it you're free to, can even be a second PR)

Reply:
@TheLastProject - thanks for the feedback. Let me finish working on the changes here and I think the upgraded flow can be part of a different PR.

Reply:
@TheLastProject, here is a gif recording of the "starting outside" flow:
![qemu-system-x86_64_61xmebUzo3](https://user-images.githubusercontent.com/23402988/196536741-724ff478-5ef5-4d1f-8cd9-ddd873dcb632.gif)

You will notice at the end that when a user presses the back button, the application closes since it was opened by the intent filter.

Reply:
Yes, I noticed that's the flow and that's what I worry about. The import starts with no user interaction to make sure the user meant to import the file (maybe they just accidentally clicked the wrong app).

That's why I'm asking to add step 3, to ask the user the import type so that it will work even if the file gets renamed and to add a "confirmation step":
- Catima opens and asks what type the import is (Catima/Stocard/Fidme/etc.)

> You will notice at the end that when a user presses the back button, the application closes since it was opened by the intent filter.

That seems reasonable to me. Perhaps slightly odd, but users keeping their file manager open feels slightly odd as well, so whatever, no opinion :)

> I think the upgraded flow can be part of a different PR

Rewriting the flow that existed before this PR definitely makes sense in another PR yeah (maybe easiest to have that done first). But I'd rather not merge this PR in its current state due to the risk of users accidentally overwriting some of their data and me getting lots of angry emails about it x.x